lib/wootric/index.js

```diff
@@ -0,0 +1,146 @@
+  // if it's ready immediately after `initialize()` is called you can call `this.ready()` here.
+  window._wootric = this;
+  var self = this;
+  window.wootricSettings = {
+    response_callback: this.trackSurveyResponse,
+    decline_callback: this.trackSurveyDecline,
+    survey_callback: this.trackSurvey
+  };
+
+  window.wootricSettings.account_token = this.options.account_token;
+  //Load our library, once it is loaded, we are ready
+  this.load('library', function(){
+    window.wootric("run");
+    self.ready();
+  });
+};
+
+// Track the survey result.  This is called back once the result is sent to Wootric
+// by way of the wootricSettings.response_callback callback
+Wootric.prototype.trackSurveyResponse = function(score, text){
+  this.track('NPS Response', {
+    email: window.wootricSettings.email,
+    createdAt: window.wootricSettings.created_at,
+    score: score,
+    text: text
+  })
+}
+
+// Track the survey result.  This is called back once the result is sent to Wootric
+// by way of the wootricSettings.decline_callback callback
+Wootric.prototype.trackSurveyDecline = function(){
+  this.track('NPS Decline', {
+    email: window.wootricSettings.email,
+    createdAt: window.wootricSettings.created_at
+  })
+}
+
+// Track the survey result.  This is called back once the result is sent to Wootric
+// by way of the wootricSettings.survey_callback callback
+Wootric.prototype.trackSurvey = function(){
+  this.track('NPS Survey', {
+    email: window.wootricSettings.email,
+    createdAt: window.wootricSettings.created_at
+  })
+}
+
+/**
+ * Has the Wootric library been loaded yet?
+ *
+ * @return {Boolean}
+ */
+
+Wootric.prototype.loaded = function(){
+  // TODO: fill in the logic required to check if the library has already been
+  //We are always ready since we are just setting a global variable in initialize
+  return window.wootric;
+};
+
+/**
+ * Identify a user.
+ *
+ * @param {Facade} identify
+ */
+
+Wootric.prototype.identify = function(identify){
+  //console.log("calling identify")
+  var traits = identify.traits();
+  var email = traits.trait.email;
+  var created_at = traits.trait.createdAt;
+  var unixDt = Date.parse(created_at) || +created_at;
+  if (unixDt) {
+    window.wootricSettings.created_at = unixDt;
+  }
+  window.wootricSettings.email = email;
+  //Set the rest of the traits as properties
+  window.wootricSettings.properties = traits.trait;
+  //remove the email and createdAt properties
+  delete window.wootricSettings.properties.email;
+  delete window.wootricSettings.properties.createdAt;
+  //Track page visit
+  this.page();
+  //console.log("wootricSettings", window.wootricSettings);
+  };
+
+//Store the last page name tracked
+Wootric.lastPageTracked = null;
+
+/**
+ * Page.
+ *
+ * @param {Page} page
+ */
+Wootric.prototype.page = function(page){
+  //console.log("calling page");
+  var opts = this.options;
+  //console.log(opts);
+  if (opts.account_token) {
```

Comment:
can you use the tool without an account token? would seem to me like this tool shouldn't be enabled in that setting doesn't exist (if so this check is unnecessary)